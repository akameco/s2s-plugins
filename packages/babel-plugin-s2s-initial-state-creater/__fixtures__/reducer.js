// @flow
type Count = number
type State = { count: Count, isLoading: boolean }

// $FlowFixMe
const initialState: State = {}
