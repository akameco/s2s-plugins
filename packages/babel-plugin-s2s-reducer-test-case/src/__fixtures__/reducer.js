// eslint-disable-next-line
import * as Actions from './actionTypes'

export default function(state, action) {
  switch (action.type) {
    case Actions.INCREMENT:
      return { count: state + 1 }
    case Actions.DECREMENT:
      return { count: state - 1 }
    default:
      return state
  }
}
