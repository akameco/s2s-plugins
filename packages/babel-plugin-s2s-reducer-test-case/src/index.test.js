// @flow
import path from 'path'
import pluginTester from 'babel-plugin-tester'
import plugin from '.'

const fromPath = path.resolve(__dirname, '__fixtures__', 'reducer.js')

pluginTester({
  title: 'default',
  plugin,
  snapshot: true,
  pluginOptions: { from: fromPath },
  tests: [
    {
      title: 'basic',
      code: `
import reducer, {initialState} from './actions'
    `
    },
    {
      title: 'exist test case',
      code: `
import reducer, {initialState} from './actions'
import * as actions from './actions'

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null)
})
    `
    }
  ]
})

pluginTester({
  plugin,
  tests: [
    {
      title: 'throw error',
      code: `// throw error`,
      error: /required from option/
    }
  ]
})
