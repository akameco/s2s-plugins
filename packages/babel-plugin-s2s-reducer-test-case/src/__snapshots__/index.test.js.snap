// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
"
import reducer, {initialState} from './actions'

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';

test('handle INCREMENT', () => {
  expect(actions.increment()).toEqual(null);
});

test('handle DECREMENT', () => {
  expect(actions.decrement()).toEqual(null);
});
"
`;

exports[`exist test case 1`] = `
"
import reducer, {initialState} from './actions'
import * as actions from './actions'

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null)
})

      ↓ ↓ ↓ ↓ ↓ ↓

import reducer, { initialState } from './actions';
import * as actions from './actions';

test('handle INCREMENT', () => {
  expect(actions.sample()).toEqual(null);
});

test('handle DECREMENT', () => {
  expect(actions.decrement()).toEqual(null);
});
"
`;
